flexboxAPIspecV2.md

```diff
@@ -0,0 +1,663 @@
+
+More on Flexbox available here: https://css-tricks.com/snippets/css/a-guide-to-flexbox/#aa-flexbox-properties
+
+
+# Remarks
+
+
+
+WinUI Offers several Layout options, such as StackLayout and Flowlayout. Usage of Flexbox Layout will behave similarly and interact by wrapping your desired elements in a Layout Control.
+
+#### Flex Containers vs Flex Items
+
+Flex Containers are elements that hold and specify the arangement of Flex Items. Containers have the following accepted properties:
+| Property | 
+|-|
+| Direction |
+Wrap|
+JustifyContent|
+AlignItems|
+AlignContent|
+
+Flex Items are elements that are children of a Flex Container. Items have the following accepted properties:
+| Property | 
+|-|
+Order|
+Grow|
+Shrink|
+Basis|
+AlignSelf|
+Unit|
+
+
+
+#### Traditional CSS Behavior vs Flexbox in WinUI
+
+Several behaviors may deviate from the exact behavior found in CSS Flexbox, to highlight one in XAML, items do not overflow their containers, instead they will be omitted from the layout.
+
+
+
+# API Pages
+
+_(Each of the following L2 sections correspond to a page that will be on docs.microsoft.com)_
+
+
+
+## Flex Direction
+
+
+The flex-direction property specifies the direction of the flexible items.
+Note: If the element is not a flexible item, the flex-direction property has no effect.
+
+
+```xml
+<controls:LayoutPanel>
+    <controls:LayoutPanel.Layout>
+        <controls:FlexboxLayout Direction="Row"/>
+    </controls:LayoutPanel.Layout>
+            <Grid>
+                <TextBlock>1</TextBlock>
+            </Grid>
+            <Grid>
+                <TextBlock>2</TextBlock>
+            </Grid>
+            <Grid>
+                <TextBlock>3</TextBlock>
+            </Grid>
+            <Grid>
+                <TextBlock>4</TextBlock>
+            </Grid>
+</controls:LayoutPanel>
+```
+
+
+
+### Output of the above code (styling omitted)
+![image](./FlexDirectionRow.PNG)
+
+
+
+##  Flex Direction Accepted Values
+
+| Value | Description |
+|-|-|
+| Row (Default) | The flexible items are displayed horizontally, as a row |
+| RowReverse | Same behavior as Row, but elemnts are shown in reverse |
+| Column | The flexible items are displayed vertically, as a column |
+| ColumnReverse | Same behavior as Column, but elemnts are shown in reverse |
+
+# API Details
+
+```c# (but really MIDL3)
+
+runtimeclass FlexboxLayout;
+
+    enum FlexboxDirection
+        {
+            Row = 0,
+            RowReverse = 1,
+            Column = 2,
+            ColumnReverse = 3
+        }
+```
+
+## Flex Wrap
+
+The flex-wrap property specifies whether the flexible items should wrap or not.
+
+Note: If the elements are not flexible items, the flex-wrap property has no effect.
+
+```xml
+
+<controls:LayoutPanel MaxWidth="500">
+    <controls:LayoutPanel.Layout>
+        <controls:FlexboxLayout Direction="Row" Wrap="Wrap"/>
+    </controls:LayoutPanel.Layout>
+            <Grid MinWidth="10">
```

Comment:
Again can leave out the Grids here. I'd make the content of the TextBlock be the value of the MinWidth, just to make the picture even easier to read (it's already good).

Reply:
I think I would need to manually re-do all of the example photos as they have the same styling scheme, which requires the grid for a background color